### ISSUE TITLE
not working when browser not support Symbol

### ISSUE THREAD

**Issue by ariesjia** [agraboso/redux-api-middleware#22]
if browser not support Symbol  ，the action.key is string not equal to object   ， so push error  and not send api

```
for (let key in action) {
    if (key !== [CALL_API]) {
      validationErrors.push(`Invalid root key: ${key}`);
    }
  }
```

I fount if browser support  Symbol ,  do not enter this `for` loop .  because action.key is real Symbol

Maybe this validate is not necessary .

**Comment by olslash** [comment]
+1 -- having this issue in phantom, even though it's going through babel first. Any ideas?

**Comment by olslash** [comment]
as a hotfix in my codebase, i fixed this by:
- changing CALL_API to a string ("redux-api-middleware/CALL_API") in CALL_API.js
- taking out the array around CALL_API here: https://github.com/agraboso/redux-api-middleware/blob/master/src/validation.js#L89. I think that must have been a typo.

**Comment by riophae** [comment]
You could inject a polyfill into your project.
See: https://github.com/zloirock/core-js#commonjs

**Comment by olslash** [comment]
I changed my test setup to not include Phantom so I can't verify this, but I believe I did have Symbol polyfilled. It's possible the issue was with the polyfill, or that it wasn't working properly, although other ES6 stuff like Promise did. Once i get browser tests running again I'll report back.

**Comment by riophae** [comment]
@olslash Did you bundle the project when testing in Phantom.js? As far as I know, If you didn't use something like Browserify, Phantom.js could work improperly with core.js polyfills.

**Comment by jamesbillinger** [comment]
+1  I'm having this issue in IE11.  I added es6-symbol and verified that I can create and use Symbols, but I still can't get [CALL_API] to work in IE.

**Comment by stinju** [comment]
FWIW, I ran into the same issue when using with react-native on pre-iOS-9 devices.

**Comment by danny-andrews** [comment]
I don't think we should be using symbols in the action anyway, as the redux docs suggest against it: "We recommend that you use strings and not Symbols for action types, because strings are serializable, and by using Symbols you make recording and replaying harder than it needs to be." Although this is talking specifically about action types, the same principle applies for all parts of an action.

So I think we should change the lines (here)[https://github.com/agraboso/redux-api-middleware/blob/master/src/CALL_API.js#L8-L10] from:

```javascript
const CALL_API = Symbol('Call API');

export default CALL_API;
```

to `export default 'CALL API';`

**Comment by shinnn** [comment]
@agraboso Would you accept a pull request to change `Symbol('Call API')` to `'CALL API'` as @danny-andrews suggested in https://github.com/agraboso/redux-api-middleware/issues/22#issuecomment-170611152?

**Comment by geminiyellow** [comment]
@agraboso

 i found the Discussion about use Symbol as action type.

https://github.com/rackt/redux/pull/114#issuecomment-112608830
https://github.com/rackt/redux/pull/127

and @gaearon change Symbol to prefix `@@` as initial action special type in redux.
so how about change `Symbol('Call API')` to `redux-api-middleware/API` or `@@redux-api-middleware/API` or something else.

**Comment by ariesjia** [comment]
+1 for  '@@'

**Comment by barrystaes** [comment]
Agreed. Its also what [`react-router-redux`](https://github.com/rackt/react-router-redux/) (formerly known as `redux-simple-router`) does: `@@router/UPDATE_LOCATION`

**Comment by barrystaes** [comment]
@geminiyellow 
I guess now the hard part is naming things, and do a PR.. and update docs and tests.

I think CALL_API is fine, but what about using `@@redux-api-middleware/RSAA` as the value, and `RSAA` as the value's name? This would make it somewhat self-explaining and future proof.

The example would then become
```
import { RSAA } from `redux-api-middleware`;

{
  [RSAA]: {
    endpoint: 'http://www.example.com/api/users',
    method: 'GET',
    types: ['REQUEST', 'SUCCESS', 'FAILURE']
  }
}
```

Just a thought..

**Comment by geminiyellow** [comment]
@georgebonnr 
yes, i think so, `CALL_API` is good enough, but `RSAA` is better.
`[CALL_API]` in `redux-api-middleware` is a type name,
so nouns `RSAA` looks more self-explaining, and shorter, and no `_`.
i dont like `_` :trollface:

**Comment by georgebonnr** [comment]
1. :+1:  this sounds good to me.
2. Did you mean to @ mention me in this thread?  Did my username get autofilled in because I've forked?
3. This project is rad, btw, and I'm loving it in production at Patreon.  I've been able to build some utils on top of this that we might share back soon.  :beers:

**Comment by geminiyellow** [comment]
oh **My fault**, sorry @georgebonnr .
i just want to reply @barrystaes 's https://github.com/agraboso/redux-api-middleware/issues/22#issuecomment-182865131

**Comment by stepan-romankov** [comment]
+1 It's nice to have it working on IE8

**Comment by barrystaes** [comment]
@georgebonnr I'm looking forward to your experiences and utils. :)
If its opensource, i would be happy to link these under a "Projects that use redux-api-middleware" section in the README.md file.

**Comment by jackyon** [comment]
so what is the best solution for fixing this issue?

**Comment by barrystaes** [comment]
This solution wil be included in the [next branch](https://github.com/agraboso/redux-api-middleware/tree/next) for v2.0.0-beta.

**Comment by jackyon** [comment]
@barrystaes 

thx for reminding, can't wait for the next release version.

**Comment by gilbarbara** [comment]
@barrystaes Hey!
Can you publish the `next` branch to npm @next?

[s]

**Comment by jackyon** [comment]
+1

**Comment by barrystaes** [comment]
I can not publish to npm, but @agraboso can.

**Comment by gilbarbara** [comment]
@barrystaes 
I know this but there's no `/lib` in the git repo, so it will not work (I've tried..). I'm using a local compiled version but it's lame.

@agraboso Can you please publish to npm or something? Thanks

**Comment by gaearon** [comment]
Lol, I feel bad for putting a symbol there in the first place in the `real-world` example. 😄

**Comment by gilbarbara** [comment]
ok, let's blame @gaearon 😁

**Comment by agraboso** [comment]
I just published v1.0.2 and v2.0.0-beta.1 to npm.

**Comment by abimelex** [comment]
I still get the error using version 1.0.2 is this already released there? What would be the best solution to get that fixed?
